src/UseCases/BookPayment/BookPaymentUseCase.php

```diff
@@ -52,4 +59,15 @@ public function bookPayment( int $paymentId, array $transactionData ): SuccessRe
 
 		return new SuccessResponse();
 	}
+
+	/**
+	 * @param Payment $payment
+	 * @param array<string,mixed> $transactionData
+	 *
+	 * @return VerificationResponse
+	 */
+	private function validateWithExternalService( Payment $payment, array $transactionData ): VerificationResponse {
```

Comment:
I like how this method bundles the 2 steps - creating the service and calling it. This makes the calling code much nicer to read